### DISCUSSION TITLE
Best settings for 144hz display?

### DISCUSSION THREAD

**Discussion by almeath** [dosbox-staging/dosbox-staging#3974] [Q&A] [answered]
I recently built a new PC and was confronted with this error after migrating my exsting DOSBox Staging configuration.

![Screenshot 2024-10-05 145116](https://github.com/user-attachments/assets/727eb9cf-735d-406e-bef3-53eaea6eb12f)

Now that I finally have a 4k display that can operate at more than 60hz, I am not sure what I should do to get the most out of my setup.

As for the error message.. my drivers are 100% up-to-date. I am using an RX 6900 XT under Windows 11.

**Comment by johnnovak** [comment]
@almeath Configuration please. Also, try with the stock settings (launch with `--noprimaryconf` from a folder that has no local `dosbox.conf`).

**Reply by almeath** [reply]
Using `--noprimaryconf`  avoids any errors:

![Screenshot 2024-10-05 162709](https://github.com/user-attachments/assets/18a8ce81-726b-4bc2-88be-06d304867784)

My configuration file:

[dosbox-staging.zip](https://github.com/user-attachments/files/17265560/dosbox-staging.zip)

**Reply by johnnovak** [reply, marked as answer]
@almeath Yeah, `vsync = on` was your problem. The error message even tells you that 😄 Is it all good now?

You chopped of the end of the log—I can't see what presentation mode is auto-picked, etc.

**Reply by almeath** [reply]
Apologies, last bit was:

![Screenshot 2024-10-05 164012](https://github.com/user-attachments/assets/c5bf112a-9acb-4981-bd4f-5e61cced8fcf)

Yes, setting vsync to 'auto' has avoided the error with my existing configuration, thanks.

Is there anything else I  need to tweak or does the refresh rate optimization all happen automatically? I think I confused myself by looking at DOSBox-X where specific refresh rates could be set manually. If DOS games often ran at 75hz I presume my monitor being set to 144hz in Windows will result in some smoother framerates or scrolling?

**Reply by johnnovak** [reply]
Yeah, forget about the refresh rate config stuff in all other DOSBox variants, including X...

IMO the best way to test is with the intro of Pinball Dreams. If the vertical credits scroller in the intro is 100% butter-smooth (the white text on black background scroller), you're good.

Is your monitor VRR? If yes, I think you're done, the defaults should work for you. People in the team have spent considerable time to make the defaults work fine in most common setups (VRR, fixed 60 Hz, various OSes, etc.)

**Reply by almeath** [reply]
Yes, I have an Eve Spectrum S073D09 which is described as being VRR compliant, supporting G-Sync and FreeSync. So it would appear I am all set. I will get a hold of Pinball Dreams and check it out, thanks. :-)

**Reply by johnnovak** [reply]
Great, please confirm whether the defaults give you perfect results like I'm thinking they would. In general, don't enforce vsync in the Staging config, neither at the driver level on VRR monitors to get the best results (perfect sync & perfectly smooth scrolling).

**Reply by almeath** [reply]
Thanks. I think I should be good but will report back to confirm. 

The only settings I can find in my driver app referring to anything with 'sync' is "FreeSync", which has always been enabled since I built the PC.

![Screenshot 2024-10-05 171536](https://github.com/user-attachments/assets/66a79ead-4f70-4fd5-8ed0-91451f018a2d)

There is also "wait for vertical refresh" and "enhanced sync" both of which are off by default, so I will leave them that way.

**Reply by almeath** [reply]
I tested some games that were affected by scrolling lag and jerkiness on my old 60hz display, such as Epic Pinball, Lemmings 2, and Zool. All of them are now scrolling exceptionally smoothly and amazingly fast.

Thanks for the help today and for all the great work the team is doing - looking forward to the full release of 0.82!

**Reply by johnnovak** [reply]
@almeath Cheers, that's great to hear Staging picks the best defaults on your setup to achieve perfect vsync! 🥳 0.82.0 should be out in a week or two, hope you'll enjoy it, man 😎
